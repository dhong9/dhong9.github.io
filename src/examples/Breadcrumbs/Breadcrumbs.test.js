import renderer from "react-test-renderer";
import Breadcrumbs from "examples/Breadcrumbs";

// Mocks
jest.mock("components/MKBox", () => {
    const { forwardRef } = jest.requireActual("react");
    return {
        __esModule: true,
        default: forwardRef(() => <div>Mock Box</div>)
    };
});
jest.mock("components/MKTypography", () => {
    const { forwardRef } = jest.requireActual("react");
    return {
        __esModule: true,
        default: forwardRef(() => <p>Mock Typography</p>)
    };
});
jest.mock("@mui/material/Breadcrumbs", () => {
    const { forwardRef } = jest.requireActual("react");
    return {
        __esModule: true,
        default: forwardRef(() => <div>MUI Breadcrumbs</div>)
    };
});

describe("BaseLayout", () => {
    it("renders", () => {
        const component = renderer.create(
            <Breadcrumbs routes={[{ label: "Bread", route: "/crumbs" }]}>
                Test Breadcrumbs
            </Breadcrumbs>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});