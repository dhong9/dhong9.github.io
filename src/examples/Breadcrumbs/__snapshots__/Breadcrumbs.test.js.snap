// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs renders with route 1`] = `
<div>
  Mock Box
</div>
`;

exports[`Breadcrumbs renders without route 1`] = `
<div>
  Mock Box
</div>
`;
